Content.Server/Construction/FlatpackSystem.cs

```diff
@@ -33,11 +33,16 @@ private void OnStartPack(Entity<FlatpackCreatorComponent> ent, ref FlatpackCreat
         if (!_itemSlots.TryGetSlot(uid, comp.SlotId, out var itemSlot) || itemSlot.Item is not { } machineBoard)
             return;
 
-        if (!TryComp<MachineBoardComponent>(machineBoard, out var boardComp))
-            return;
-
-        if (!MaterialStorage.CanChangeMaterialAmount(uid, GetFlatpackCreationCost(ent, (machineBoard, boardComp))))
-            return;
+        if (TryComp<MachineBoardComponent>(machineBoard, out var boardComp))
+        {
+            if (!MaterialStorage.CanChangeMaterialAmount(uid, GetFlatpackCreationCost(ent, (machineBoard, boardComp))))
+                return;
+        }
+        else
+        {
+            if (!MaterialStorage.CanChangeMaterialAmount(uid, GetFlatpackCreationCostForComputer(ent)))
+                return;
+        }
```

Comment:
you should do an explicit trycomp for `ComputerBoardComponent` here, as it leaves the intent kinda ambiguous. Also, add a return if it can't get either comp. It won't really matter but it makes it clearer that the entity should have one or the other.